packages/@aws-cdk/aws-lambda/lib/log-retention.ts

```diff
@@ -58,13 +65,69 @@ export class LogRetention extends cdk.Construct {
 
     // Need to use a CfnResource here to prevent lerna dependency cycles
     // @aws-cdk/aws-cloudformation -> @aws-cdk/aws-lambda -> @aws-cdk/aws-cloudformation
-    new cdk.CfnResource(this, 'Resource', {
+    const resource = new cdk.CfnResource(this, 'Resource', {
       type: 'Custom::LogRetention',
       properties: {
         ServiceToken: provider.functionArn,
         LogGroupName: props.logGroupName,
         RetentionInDays: props.retention === Infinity ? undefined : props.retention
       }
     });
+
+    this.logGroupName = resource.getAtt('LogGroupName').toString();
+    // Append ':*' at the end of the ARN to match with how CloudFormation does this for LogGroup ARNs
+    // See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html#aws-resource-logs-loggroup-return-values
+    this.logGroupArn = cdk.Stack.of(this).formatArn({
+      service: 'logs',
+      resource: 'log-group',
+      resourceName: this.logGroupName,
+      sep: ':'
+    }) + ':*';
```

Comment:
why not just append `:*` to the `logGroupName`:

```ts
formatArn({
  service: 'logs',
  resource: 'log-group',
  resourceName: `${this.logGroupName}:*`,
  // ...
})
```

Reply:
Good idea. It's cleaner.